src/App.jsx

```diff
@@ -14,33 +15,99 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_FIELD_ALPHABETICALLY = 'Sort alphabetically';
+const SORT_FIELD_LENGTH = 'Sort by length';
+const SORT_FIELD_REVERSE = 'Reverse';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function getPreperedGoods(sortField, checkIsReverse) {
+  const preperedGoods = [...goodsFromServer];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (sortField) {
+    preperedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_ALPHABETICALLY:
+          return good1.localeCompare(good2);
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+        case SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (checkIsReverse) {
+    preperedGoods.reverse();
+  }
+
+  return preperedGoods;
+}
+
+export const App = () => {
+  const [sortField, setsortField] = useState('');
+  const [checkIsReverse, setcheckIsReverse] = useState(false);
+
+  const visibleGoods = getPreperedGoods(sortField, checkIsReverse);
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() => setsortField(SORT_FIELD_ALPHABETICALLY)}
+          type="button"
+          className={
+            sortField === SORT_FIELD_ALPHABETICALLY
+              ? 'button is-info'
+              : 'button is-info is-light'
+          }
+        >
+          {SORT_FIELD_ALPHABETICALLY}
+        </button>
+
+        <button
+          onClick={() => setsortField(SORT_FIELD_LENGTH)}
+          type="button"
+          className={
+            sortField === SORT_FIELD_LENGTH
+              ? 'button is-success'
+              : 'button is-success is-light'
+          }
+        >
+          {SORT_FIELD_LENGTH}
+        </button>
+
+        <button
+          type="button"
+          onClick={() => setcheckIsReverse(!checkIsReverse)}
+          className={
+            checkIsReverse ? 'button is-warning' : 'button is-warning is-light'
+          }
+        >
+          {SORT_FIELD_REVERSE}
+        </button>
+
+        {(sortField || checkIsReverse) && (
```

Comment:
Create a variable for this condition `sortField || checkIsReverse` to avoid combining different logical operators in the same time